rfcs/01929_tls_attestation.md

```diff
@@ -0,0 +1,258 @@
+This assures that Exported Keying Material is signed by the TEE platform. By
+sending the TEE report over the TLS session, we cryptographically bind the
+session to the TEE platform.
+
+## Attestation
+
+Since Exported Keying Material is derived from the TLS master secret, that is
+shared by both participants of the TLS session, this scheme supports
+**bi-directional remote attestation**. Which means that both client and server
+can run inside the TEE environment and remotely attest each other. Remote
+attestation is performed by checking that:
+
+- TEE reports were generated by valid TEE platform
+- TEE reports contain correct Exported Keying Material derived from this
+  particular TLS session
+
+It’s also important to note that this scheme can work with any combination of
+client and server being in a TEE or not (i.e. also supports **unidirectional
+remote attestation**).
+
+### Attestation Workflow
+
+This section presents a high-level attestation workflow overview. The workflow
+of the remote attestation involves 3 interacting entities:
+
+- **Server**
+  - Server application that processes requests from TLS clients
+    - Can run inside the TEE environment
+- **Client**
+  - Client application that connects to applications using TLS
+    - Can run inside the TEE environment
+- **Provider**
+  - TEE platform provider that can confirm authenticity of TEE platform keys
+  - It’s important to note that the Provider is an external server (i.e. belongs
+    to AMD)
+
+It is assumed that prior to the attestation workflow both **Server** and
+**Client** remotely attest their TEE platform keys with the **Provider**.
+**Provider** sends back signatures rooted in some well-known root key (i.e. AMD
+root key), thus confirming the authenticity of TEE platform keys.
+
+Another important note is that the current version of the attestation protocol
+prohibits
+[TLS session resumption](https://tools.ietf.org/html/rfc8446#section-2.2),
+because the attestation takes place on the application level and it would need
+to be redone once the TLS session is resumed.
+
+#### Token Derivation
+
+Exported Keying Material is derived using
+[HKDF](https://tools.ietf.org/html/rfc8446#section-7.5) from the TLS session
+master secret and a set of parameters:
+
+- _Label_
+  - Label is required to disambiguate different applications that use the same
+    Exported Keying Material mechanism
+- _Context_
+  - Allows the application to mix its own data the pseudo-randomly generated
+    Exported Keying Material
+- _Length_
+  - Bit length of the generated value
+
+It’s important to note that with a fixed set of parameters Exported Keying
+Material value for a given TLS session stays the same - i.e. it’s a
+deterministic algorithm. Thus if both client and server negotiate corresponding
+parameters (or they are constant), they can derive equal Exported Keying
+Material values.
+
+Exported Keying Material is prefixed with the purpose string
```

Comment:
Unclear what this means; AFAICT it is not true that the exported keying material is prefixed with this; rather, it must be prefixed with this before being used